code/modules/reagents/reagent_dispenser.dm

```diff
@@ -203,8 +203,10 @@
 		explosion(src.loc,1,2,4)
 	else if (reagents.total_volume > 100)
 		explosion(src.loc,0,1,3)
-	else
+	else if (reagents.total_volume > 0)
```

Comment:
This will explode if it's full of water and has no fuel in it at all. You should change these checks to look for fuel instead of just any contents. I do think it was the intended feature that a <100 fuel tank would still have a small non-dev explosion due to fumes.

The best possible implementation of this would be dispenser-level (or even obj level!) and scale the explosion based on amount of fuel in the tank (instead of amount of any reagent in the tank)

Reply:
While you're right, this proc currently won't ever trigger unless a different proc (currently all under the fueltank object) calls explode(), and there's no way outside of admin fuckery to fill fuel tanks with any reagent other than welding fuel. Moving it up to the dispenser level isn't a bad idea. I'm actually working on a larger reagent dispenser overhaul that should allow for refilling tanks with arbitrary reagents, but I'm not sure how that'll take and wanted to get this out of the door as a quick bugfix in the meantime.